### PR TITLE
Fix : #13694 NoneType in productChannels and removeVariants in productChannelListingUpdate

### PR DESCRIPTION
**Pull request by Manoj-gowra**
I want to merge this change because NoneType in productChannels and removeVariants in productChannelListingUpdate is fixed

Fixes : #13694

# Pull Request Checklist

- [x] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/channel/mutations/base_channel_listing.py [hunk 1]
```diff
@@ -66,8 +66,11 @@ def clean_channels(
         input_source="add_channels",
     ) -> Dict:
         add_channels = input.get(input_source, [])
-        add_channels_ids = [channel["channel_id"] for channel in add_channels]
+        add_channels_ids = [channel["channel_id"] for channel in add_channels
+                            if channel.get('remove_variants',[]) is not None]
```
Comment (8r2y5) [inline saleor/graphql/channel/mutations/base_channel_listing.py:70]: Providing default value for key that is not present in the dict is not necessary in this case
```suggestion
                            if channel.get('remove_variants')]
```

Comment (fowczarek) [reply saleor/graphql/channel/mutations/base_channel_listing.py:70]: In the current implementation if you want to add a new channel with variables like eg.
```
{
    "id": "UHJvZHVjdDoxNTI=",
    "input": {
      "removeChannels": null,
      "updateChannels": [
        {
          "channelId": "Q2hhbm5lbDoy",
          "isAvailableForPurchase": true,
          "isPublished": true,
          "removeVariants": null,
          "visibleInListings": true
        }
      ]
    }
  }
```
Then no channel was created.

In any situation when `removeVariants` is empty causes a situation when the whole object is deled from `updateChannels` list.

**Review by fowczarek — DISMISSED** [review]

File: saleor/graphql/channel/mutations/base_channel_listing.py (hunk 1 above)
Comment (Manoj-gowra) [reply saleor/graphql/channel/mutations/base_channel_listing.py]: Apologies I wrongly understood the issue the first time. As @fowczarek stated it shouldn't behave like that.
Fixed the mistake now. Please check the new commit.

**Comment by Manoj-gowra** [comment]
Apologies I wrongly understood the issue the first time. As @fowczarek stated it shouldn't behave like that.
Fixed the mistake now. Please check the new commit.

**Review by korycins — APPROVED** [review]

**Review by fowczarek — COMMENTED** [review]
Could you also add a test for empty `removeVariants`?

**Comment by Manoj-gowra** [comment]
Added test cases for empty `removeVariants` and ran pre-commit which is blocking the merge earlier. Kindly review and let me know if anything is needed.

**Review by IKarbowiak — APPROVED** [review]

**Review by szczecha — APPROVED** [review]

File: saleor/graphql/product/tests/mutations/test_product_channel_listing_update.py [hunk 2]
```diff
@@ -1222,6 +1254,39 @@ def test_product_channel_listing_remove_variant_as_app(
     assert len(variant.channel_listings.all()) == 1
 
 
+def test_product_channel_listing_remove_variant_is_None_as_app(
+    app_api_client, product, permission_manage_products, channel_USD, channel_PLN
+):
+    # given
+    variant = product.variants.first()
+    print(variant)
```
Comment (fowczarek) [inline saleor/graphql/product/tests/mutations/test_product_channel_listing_update.py:1262]:
```suggestion
```

**Comment by Manoj-gowra** [comment]
When I ran the pytest locally, it was successful. However, it's failing here. Can anyone provide guidance? Is it possible that the failure is due to the changes I made to the debug print in GitHub without applying them locally beforehand?
